src/preamble.js

```diff
@@ -843,7 +843,21 @@ var splitModuleProxyHandler = {
       var imports = {'primary': Module['asm']};
       // Replace '.wasm' suffix with '.deferred.wasm'.
       var deferred = wasmBinaryFile.slice(0, -5) + '.deferred.wasm'
-      instantiateSync(deferred, imports);
+      if (Module["loadSplitModule"]) {
+        try {
+          // returns [instance, module]
+          var moduleInfo = Module["loadSplitModule"](prop, deferred, imports);
+          if (!moduleInfo || !moduleInfo[0]) {
+              throw "loadSplitModule handler failed!";
+          }
+        } catch(e) {
+            var str = e.toString();
+            err('failed to compile wasm module: ' + str);
+            throw e;
+        }
```

Comment:
```suggestion
        if (!Module["loadSplitModule"](prop, deferred, imports)) {
          abort('failed to load split module');
        }
```
We could simplify this by just having the handler return success or failure.

Reply:
i wanted to keep the return value same as instantiateSync, this is better if we do not need the instance and module information.